backend/src/main/java/gov/cdc/usds/simplereport/config/authorization/DemoAuthenticationConfiguration.java

```
@@ -0,0 +1,153 @@
+            LOG.trace("Parsing authorization header [{}]", authHeader);
+            String userName = authHeader.substring(DEMO_AUTHORIZATION_FLAG.length());
+            securityContext.setAuthentication(
+                new TestingAuthenticationToken(userName, null, List.of()));
+          }
+          chain.doFilter(request, response);
+        };
+    return new FilterRegistrationBean<>(filter);
+  }
+
+  @Bean
+  public AuthorizationService getDemoAuthorizationService(DemoUserConfiguration config) {
+    return new DemoUserAuthorizationService(config);
+  }
+
+  @Bean
+  public IdentitySupplier getDemoIdentitySupplier(DemoUserConfiguration config) {
+    return new DemoUserIdentitySupplier(config);
+  }
+
+  /**
+   * Creates and returns a {@link Supplier} that examines the current security context, finds the
+   * username of the current user, and returns the {@link DemoUser} that matches that username. If a
+   * default user has been configured, that user will be returned in the event that there is no
+   * username or the username does not match any known DemoUser.
+   *
+   * @param config the configured list of DemoUsers, with or without a default user set.
+   */
+  private static Supplier<Optional<DemoUser>> getCurrentDemoUserSupplier(
+      DemoUserConfiguration config) {
+    return () -> {
+      Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+      String userName = null;
+      if (auth == null) { // have to allow defaulting to work even when there is no authentication
+        LOG.warn("No authentication found: current user will be default (if available)");
+      } else if (auth.isAuthenticated()) {
+        userName = auth.getName();
+      } else {
+        LOG.info("Unauthenticated user in demo mode: current user will be default (if available)");
+      }
+      DemoUser found = config.getByUsername(userName);
+      if (found == null) {
+        LOG.error("Invalid user {} in demo mode", userName);
```

i think this should throw an exception, not return a null value below

---

That's probably true, there's no valid use case for misspelling the e-mail in the auth token. 🙃